hip-service/Link/Patient/LinkPatient.cs

```diff
@@ -0,0 +1,116 @@
+                    (null, new Error(ErrorCode.CareContextNotFound, "Care context not found for given patient"));
+            }           
+            
+            var linkRefNumber = Guid.NewGuid().ToString();
+            
+            // method call for generating the OTP
+            var session = new Session(linkRefNumber, new Communication(IdentifierType.MOBILE,""));
+
+            if (await _patientVerification.GenerateVerificationToken(session) != null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, Error>
+                    (null, new Error(ErrorCode.OtpInValid, "Unable to create token"));
+            }
+            
+            var (_, exception) = await linkPatientRepository.SaveLinkPatientDetails(linkRefNumber, patient.ConsentManagerId,
+                patient.ConsentManagerUserId, patient.ReferenceNumber, patient.CareContexts
+                    .Select(context => context.ReferenceNumber).ToArray());
+
+            if (exception != null)
+            {
+                return new Tuple<PatientLinkReferenceResponse, Error>
+                    (null, new Error(ErrorCode.CareContextNotFound, "Unable to store data to Database"));
+            }
+            
+            var expiry = new DateTime (DateTime.Today.Year, DateTime.Today.Month, DateTime.Today.Day, 
+                    DateTime.Now.Hour, DateTime.Now.Minute+1, DateTime.Now.Second).ToUniversalTime().
+                ToString("yyyy-MM-ddTHH:mm:ssZ"); 
+
+            var meta = new LinkReferenceMeta(nameof(IdentifierType.MOBILE).ToUpper(), 
+                _patientRepository.GetPatientInfoWithReferenceNumber(patient.ReferenceNumber).PhoneNumber, expiry);
+            var patientLinkReferenceResponse = new PatientLinkReferenceResponse(new LinkReference(linkRefNumber,"MEDIATED",meta));
+
+            return new Tuple<PatientLinkReferenceResponse, Error>(patientLinkReferenceResponse, null);
+        }
+
+        public async Task<Tuple<PatientLinkResponse, Error>> VerifyAndLinkCareContext(PatientLinkRequest request)
+        {
+            if (await _patientVerification.AuthenticateVerificationToken(request.LinkReferenceNumber, request.Token) != null)
+            {
+                return new Tuple<PatientLinkResponse, Error>
+                    (null, new Error(ErrorCode.OtpInValid, "Otp Invalid"));   
+            }
+
+            var (linkRequest, exception) =
+                await linkPatientRepository.GetPatientReferenceNumber(request.LinkReferenceNumber);
+            if (exception != null)
+            {
+                return new Tuple<PatientLinkResponse, Error>
+                    (null, new Error(ErrorCode.NoPatientFound, "No request found"));
+            }
+
+            var patientInfo = _patientRepository.GetPatientInfoWithReferenceNumber(linkRequest.PatientReferenceNumber);
+            var patient = _patientRepository.GetPatientInfoWithReferenceNumber(linkRequest.PatientReferenceNumber);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'patient'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4769144)

Reply:
Done